src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/service/SupplierPricingService.kt

```diff
@@ -106,11 +107,15 @@ class SupplierPricingService(
 
   private fun getLocationBy(agencyId: String): Location? = locationRepository.findByNomisAgencyId(sanitised(agencyId))
 
-  fun priceHistoryForJourney(supplier: Supplier, fromAgencyId: String, toAgencyId: String) =
-    auditService.auditEventsByType(AuditEventType.JOURNEY_PRICE)
+  fun priceHistoryForJourney(supplier: Supplier, fromAgencyId: String, toAgencyId: String): Set<AuditEvent> {
+    val sanitisedFrom = sanitised(fromAgencyId)
+    val sanitisedTo = sanitised(toAgencyId)
+
+    return auditService.auditEventsByType(AuditEventType.JOURNEY_PRICE)
       .associateWith { PriceMetadata.map(it) }
-      .filterValues { it.supplier == supplier && sanitised(it.fromNomisId) == sanitised(fromAgencyId) && sanitised(it.toNomisId) == sanitised(toAgencyId) }
+      .filterValues { it.supplier == supplier && it.fromNomisId == sanitisedFrom && it.toNomisId == sanitisedTo }
```

Comment:
So this would happen loads of times given there are 25000 + price audit events.  We don't need to sanitise each row.